### PR TITLE
Document migration facilities

### LINKED ISSUE #107: Document migration facilities
Deliverables:

- Document the general facilities, via "corresponding quantity".
- Migrate our nholthaus/units conversion file, including the unit tests.
- Add documentation page using nholthaus as worked migration example.

### PR DESCRIPTION
**Pull request by chiphogg**
This PR adds how-to guides for the migration facilities.  We include a
general how-to guide, and a specific guide for the built-in migration
facilities for the nholthaus library.  We also add some more link
anchors in other docs as needed, and remove the howto stub from the
`corresponding_quantity.md` reference page.

Fixes #107.

### CODE REVIEW

File: docs/howto/interop/index.md [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+        expect_equivalent<::units::data::byte_t>(bytes);
+    }
+    ```
+
+    Note how the utility makes these very easy to read.
+
+### Implement the Definitions
+
+The general strategy for writing an implementation that can make these unit tests pass includes
+these steps.
```
Comment (johnzoppina) [inline docs/howto/interop/index.md:118]: Read the above sentence aloud -- it's a mouthful. How about:

"Generally, writing an implementation that passes these unit tests includes:

1. Figuring out...
2. Writing...
3. Writing..."

...It changes the imperative verbs in the list to gerunds, which contradicts my normal advice, but for the sake of making the lead-in sentence a little clearer, I think it's worth it.

File: docs/howto/interop/nholthaus.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+avoided.  _However_, it can be done safely if we confine this dependence to a single file --- the
+_shim_ --- and provide _only that file_ to our end users.  This shim must do the following three
+things, in this order.
+
+1. Include the nholthaus-relevant parts of Au (in whatever manner is appropriate for your project).
+2. Include the nholthaus units library (in whatever manner is appropriate for your project).
+3. Include your copy of `compatibility/nholthaus_units.hh`, **after** every other file.
+    - (The include path comes from Step 1 above.)
+
+We have included a [self-explaining example
```
Comment (johnzoppina) [inline docs/howto/interop/nholthaus.md:42]: "Include" seems to have shifted in meaning between the previous paragraph/steps and _this_ paragraph -- is that fair? If that's true, maybe we shift "include" to "provide" in this paragraph.

**Review by johnzoppina — COMMENTED** [review]
@chiphogg Take a look at a couple of the comments here. I'm not requesting changes per se because these are matters of taste more than anything else.

I've made a couple changes directly -- heading styles should follow sentence case, except for page titles, and I'm fussy about that/which.

File: docs/howto/interop/index.md [hunk 3]
```diff
@@ -0,0 +1,295 @@
+# Inter-library Interoperation
+
+This guide explains how use Au's [corresponding quantity
+machinery](../../reference/corresponding_quantity.md) to set up minimum-friction interoperation
+between Au and some other C++ units library.  The main use case is for migrating either to or from
+Au.  Once you set this up, you will be able to pass Au's quantity types to APIs expecting equivalent
+types from the other library, and vice versa.  This flexibility will make that migration much
+easier, because it will let you migrate individual build targets without forcing you to upgrade all
+of their callers at the same time.
```
[the remark this answers is not in the document]
Comment (chiphogg) [reply docs/howto/interop/index.md:9]: Thanks!  I ended up going with option 2.  (Option "to"?)

File: docs/howto/interop/index.md [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+
+Here are the steps involved.
+
+1. Define the equivalences.
+2. _(Optional)_ Create a shim file.
+
+!!! warning
+    A common theme in this generic guide is that the details for each step will **strongly** depend
```
[the remark this answers is not in the document]
Comment (chiphogg) [reply docs/howto/interop/index.md]: Done.

File: docs/howto/interop/index.md (hunk 1 above)
Comment (chiphogg) [reply docs/howto/interop/index.md]: Done.

File: docs/howto/interop/nholthaus.md (hunk 2 above)
Comment (chiphogg) [reply docs/howto/interop/nholthaus.md]: Great catch!  Done.

**Review by johnzoppina — APPROVED** [review]
Looks good. Thanks!

**Review by tobin — APPROVED** [review]

**Pull request merged**